src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RazorServiceFactoryBase.ExportProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Composition;
```

Comment:
Is Roslyn OOP using VS MEF? If not, should we be pulling that in? Or, is MEFv2 sufficient?

Reply:
To be clear, I would have expected us to use [CompositionHost](https://learn.microsoft.com/en-us/dotnet/api/system.composition.hosting.compositionhost) rather than the VS [ExportProvider](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.composition.exportprovider).

Reply:
Roslyn does use VS MEF, but I'm happy to change to using v2 directly. I was just copying code from Roslyn because its easier than working out how to write it myself (though perhaps the subsequent debugging to see where it went wrong actually took more time? 😛)

Reply:
Got it. If Roslyn's using VS MEF, then I have no concerns. :)